### PR TITLE
Implement fixes for several failing tests

### LINKED ISSUE #173: Fix Tests
The tests are failing after recent updates due to changes in the endpoint return types.

### PR DESCRIPTION
**Pull request by Devansh-bit**
Intends to resolve #173

### CODE REVIEW

File: controllers/project_update.go [hunk 1]
```diff
@@ -124,6 +124,16 @@ func UpdateProject(w http.ResponseWriter, r *http.Request) {
 			)
 			return
 		}
+		if tx.Error == gorm.ErrRecordNotFound {
```
Comment (harshkhandeparkar) [inline controllers/project_update.go:127]: The execution should not stop here. If the secondary mentor is not found (for example if an empty string is provided), the field should just get unset and no error should be returned.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: controllers/project_update.go (hunk 1 above)
Comment (Devansh-bit) [reply controllers/project_update.go:127]: In the [`tProjectUpdateExistent`](https://github.com/kossiitkgp/KWoC-Backend/blob/main/controllers/project_update_test.go#L80) test, 

```expectResponseJSONBodyToBe(t, res, utils.HTTPMessage{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("Secondary mentor `%s` does not exist.", projUpdateFields.SecondaryMentorUsername)})```

Expects 400 when a mentor is not found.

Comment (harshkhandeparkar) [reply controllers/project_update.go:127]: Remove that check.

File: controllers/project_update_test.go [hunk 2]
```diff
@@ -97,24 +110,13 @@ func tProjectUpdateExistent(db *gorm.DB, testUsername string, testJwt string, t
 		ReadmeLink:     "http://NewRepoLink/README",
 	}
 
-	// Test with invalid new secondary mentor
```
Comment (harshkhandeparkar) [inline controllers/project_update_test.go:100]: One last thing. Can you add a test that checks if giving an empty string unsets the mentor?

Comment (harshkhandeparkar) [reply controllers/project_update_test.go:100]: Let's do this at a later point.

**Review by harshkhandeparkar — APPROVED** [review]
LGTM

**Comment by harshkhandeparkar** [comment]
Thank you @Devansh-bit! :rocket:

**Comment by Devansh-bit** [comment]
Thanks to you as well for helping out and the immediate responses!
